### PR TITLE
Aj/copy structs

### PR DESCRIPTION
**Pull request by veriAjinkya**
Added copy_struct_to_storage intrinsic
Added allocate_memory_struct intrisinc

### CODE REVIEW

File: lib/libYulAST/Utils/FunctionSigVisitor.cpp [hunk 1]
```diff
@@ -368,6 +370,37 @@ void FunSigVisitor::fixAllocateMemArray(std::string funName) {
...
+    std::cerr
+        << "Received a non array type in reference for allocate_memory_array"
+        << std::endl;
+    std::abort();
+  }
+  allocationType = typeContext.getReferenceToType(ref->referencedType,
+                                                  YulAddressSpace::MEMORY);
+  retTypes.push_back(allocationType);
+  intrinsics.addIntrinsic(funName, argTypes, retTypes,
+                          YulIntrinsicId::ALLOCATE_MEM_ARRAY, true);
```
Comment (Technius) [inline lib/libYulAST/Utils/FunctionSigVisitor.cpp:401]: Shouldn't this use a mem struct intrinsic id or something?

File: lib/include/libYulAST/YulConstants.h [hunk 2]
```diff
@@ -224,4 +226,6 @@ const static std::string READ_FROM_MEMORY_REGEX_LIT =
     R"(^read_from_memory(.*?)$)";
 const static std::string WRITE_TO_MEMORY_REGEX_LIT =
     R"(^write_to_memory_(.*?)$)";
+const static std::string COPY_STRUCT_TO_STOTRAGE_REGEX_LIT =
+    R"(^copy_struct_to_storage_from_(.*?)_to_(.*?)$)";
```
Comment (Technius) [inline lib/include/libYulAST/YulConstants.h:230]: Why is `.*?` used? Doesn't this mean "match (zero or more optional characters)"? Why not `.*` instead?

File: lib/libYulAST/Utils/FunctionSigVisitor.cpp (hunk 1 above)
Comment (veriAjinkya) [reply lib/libYulAST/Utils/FunctionSigVisitor.cpp:401]: done

File: lib/include/libYulAST/YulConstants.h (hunk 2 above)
Comment (veriAjinkya) [reply lib/include/libYulAST/YulConstants.h:230]: it is just greedy version of .*. 
.* is used, it matches _to_ and the group 2 aswell

File: lib/libYulAST/Utils/FunctionSigVisitor.cpp (hunk 1 above)
Comment (Technius) [reply lib/libYulAST/Utils/FunctionSigVisitor.cpp:401]: Still shows as `ALLOCATE_MEM_ARRAY` in the diff here

Comment (veriAjinkya) [reply lib/libYulAST/Utils/FunctionSigVisitor.cpp]: Fixed

**Pull request merged**